examples/ct/ct_handlers_test.go

```diff
@@ -357,6 +358,48 @@ func TestAddChain(t *testing.T) {
 	assert.Equal(t, "BAEABnNpZ25lZA==", resp.Signature)
 }
 
+// Submit a chain with a valid precert but not complete path to root. Should be rejected.
+func TestAddPrecertChainMissingIntermediate(t *testing.T) {
+
+}
+
+// Submit a chain with a valid path but using a cert instead of a precert. Should be rejected.
+func TestAddPrecertChainCert(t *testing.T) {
+	km := new(crypto.MockKeyManager)
+	client := new(trillian.MockTrillianLogClient)
+
+	roots := loadCertsIntoPoolOrDie(t, []string{testonly.CACertPEM})
+	reqHandlers := CTRequestHandlers{0x42, roots, client, km, time.Millisecond * 500, fakeTimeSource}
+
+	cert, err := fixchain.CertificateFromPEM(testonly.TestCertPEM)
+	_, ok := err.(x509.NonFatalErrors)
```

Comment:
this is a type assertion, right?
so I believe you are checking that CertificateFromPEM's err is nil or an x509.NonFatalError, and will t.Fatal for anything else?